lib/middleware/sessionCache.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Checks if a session exists in cache using the session token as an id and returns the session found
+ *
+ * @params {Object} mbaasApi object which allows access to cache
+ * @params {String} sessionToken string that is used as an id for the specified session in cache.
+ * @params {Function} cb function callback
+ */
+function checkSession(mbaasApi, sessionToken, cb) {
+  var options = {
+    "act": "load",
+    "key": sessionToken
+  };
+
+  mbaasApi.cache(options, function(err, session) {
+    if (err) {
+      console.log("Error getting cache value: ", err);
+      return cb(err);
+    }
+
+    return cb(undefined, session);
+  });
+}
+
+/*
+ * Saves specified session in cache
+ *
+ * @params {Object} mbaasApi object which allows access to cache
+ * @params {String} sessionToken string that is used as an id for the specified session in cache.
+ * @params {Object} session object which holds all the session details
+ * @params {Function} cb function callback
+ */
+function saveSession(mbaasApi, sessionToken, session, cb) {
+  var options = {
+    "act": "save",
+    "key": sessionToken,
+    "value": session
```

Comment:
There needs to be an expiry to indicate when this cache entry should expire

https://access.redhat.com/documentation/en/red-hat-mobile-application-platform-hosted/3/paged/cloud-api/chapter-1-fhcache